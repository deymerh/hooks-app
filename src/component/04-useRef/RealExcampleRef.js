import React, { useState } from 'react';
import { MultipleCustomHooks } from '../03-example/MultipleCustomHooks';

export const RealExcampleCustomHook = () => {
  const [show, setShow] = useState(false);
  return (
    <div>
      <h1>RealExcampleRef</h1>
      <br />
      {show && <MultipleCustomHooks />}
      <button
        onClick={() => setShow(!show)}
        className="btn btn-primary mt-5"
      >
        Show/Hide
      </button>
    </div>
  )
}
