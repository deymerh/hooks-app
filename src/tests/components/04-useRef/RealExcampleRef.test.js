import React from 'react';
import { shallow } from 'enzyme';
import { RealExcampleCustomHook } from '../../../component/04-useRef/RealExcampleRef';

describe('Pruebas en en el componente <RealExcampleRef/>', () => {

  const wrapper = shallow(<RealExcampleCustomHook />);

  test('Debe mostrar el componente correctamente', () => {
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);
  });

  test('Debe mostrar el componente <MultipleCustomHooks/> de acuerdo al click', () => {
    wrapper.find('button').simulate('click');
    expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);
  });


});
