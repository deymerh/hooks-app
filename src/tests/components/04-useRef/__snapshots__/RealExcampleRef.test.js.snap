// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en en el componente <RealExcampleRef/> Debe mostrar el componente correctamente 1`] = `
<div>
  <h1>
    RealExcampleRef
  </h1>
  <br />
  <button
    className="btn btn-primary mt-5"
    onClick={[Function]}
  >
    Show/Hide
  </button>
</div>
`;
